### ISSUE TITLE
Universal Music Scraper : not fetching <year> for albums

### ISSUE THREAD

**Issue by Snaptags** [Komet/MediaElch#1377]
**Describe the bug**
Regardless of the source the scraper writes empty `<year>` tags into the artist.nfo

**To Reproduce**
- Download information (ctrl+f) for any artist/band

**Expected behavior**
- put the release year into the tags

**Screenshots**
![image](https://user-images.githubusercontent.com/1249745/147810437-4552d119-81e9-4d01-91cf-c5b457cb8855.png)

**MediaElch Version:**
 - [X] 2.8.13-dev (nightly)
 - [X] 2.8.12 stable

**Operating System:**
 - [X] Windows
 - [ ] macOS
 - [ ] Linux (distribution: _._)

**Comment by bugwelle** [comment]
Hi there,

thank you for filing this issue.

I tried to recreate the issue with an album from Rammstein, but the year was scraped successfully:

```xml
    <releasedate>2004-09-27</releasedate>
    <year>2004</year>
```

Scraping an artist, I also get the album+year in the discography tab and in the artist.nfo:

```xml
    <album>
        <title>6 Rough-Mixes / 3 Mixes</title>
        <year>1995</year>
    </album>
```

Also for "No Doubt":

<img width="675" alt="Screenshot 2021-12-31 at 12 35 54" src="https://user-images.githubusercontent.com/1667306/147821297-75bfb988-f7d8-4aa0-9451-21f46728f1c7.png">

I'm using TheAudioDb as my main music scraper. Which do you use? (Settings -> Scraper -> Universal Music Scraper -> Prefer)

Regards,
Andre

**Comment by Snaptags** [comment]
It's the same for all "preferred" settings :-(
I even removed all settings from the Windows Registry and my appdata folder, still no years…

**Comment by bugwelle** [comment]
Happy new year. :-)

Is anything else scraped? Do the movie scrapers work? If nothing is scraped, then it may be another issue with the scraper. 

Do you have an `advancedsettings.xml` file? (if you don't know what this is, then you don't use one. :-) )

**Comment by Snaptags** [comment]
I get all sorts of information for music, genres, the fanart, the artist biography, just the years are missing :-(
Movies do contain the release year, so this seems to be music specific. I don't have an `advancedsettings.xml`. but maybe I should? :-)

**Comment by bugwelle** [comment]
> I get all sorts of information for music, genres, the fanart, the artist biography, just the years are missing :-( Movies do contain the release year, so this seems to be music specific. I don't have an `advancedsettings.xml`. but maybe I should? :-)

Oh! Thank you very much for that detail. In that case it may actually be a bug in the UniversalMusicScraper. It may be that the year is overwritten by an "empty" (invalid) year. I'll check.

**Comment by bugwelle** [comment]
Ah, another question: Do you load the discography fresh, i.e. do you have existing albums listed in the discography of the artist?

**Comment by Snaptags** [comment]
I enabled the logfile via `advancedsettings.xml`:
```
MediaElch 2022-01-04 12:35:00.164 INFO  : [Music Scraper] Searching for artist: "No Doubt"
MediaElch 2022-01-04 12:35:02.491 WARN  : Network Error (load) "Error transferring https://www.theaudiodb.com/api/v1/json/7490823590829082posuda/artist-mb.php?i=fbd2a255-1d57-4d31-ac11-65b671c19958 - server replied: Not Found"
MediaElch 2022-01-04 12:35:02.541 WARN  : Network Error (load) "Error transferring https://www.theaudiodb.com/api/v1/json/7490823590829082posuda/discography-mb.php?s=fbd2a255-1d57-4d31-ac11-65b671c19958 - server replied: Not Found"
MediaElch 2022-01-04 12:35:03.298 WARN  : Error parsing music json:  "illegal value"
MediaElch 2022-01-04 12:35:03.298 WARN  : Error parsing music json:  "illegal value"
```

That's a bit unexpected because of
![image](https://user-images.githubusercontent.com/1249745/148053573-cad076fe-f7ca-436c-a743-7328463ec84b.png)

**Comment by bugwelle** [comment]
While I couldn't reproduce your issue (yet), I found a bug in our AllMusic scraper and added a test for it as well.
Furthermore, I improved some logic for scraping the "year" but I don't think this will fix your issue.

See #1379

**Comment by bugwelle** [comment]
I think I found the issue. The discography is loaded from Discogs. Their site is locale-aware, i.e. in Germany, you'll see

```
<td class="year has_header" data-header="Jahr: ">1991</td>
```
while MediaElch expects:
```
<td class="year has_header" data-header="Year: ">1991</td>
```

("Jahr" vs "Year")

Whether this is really the case, I don't know. My system is set to English, though my Browser (Firefox) is German. MediaElch also doesn't send an `Accept-Language` header which is why I would expect it to default to English. 

I don't know if I found the actual bug, but it may be the case. I have a German Windows installation at home, which I will try in the next few days.
